windows/cross-compile/build.sh

```diff
@@ -92,1399 +76,45 @@ then
-rm -f /etc/apt/apt.conf.d/docker-clean
-
-readonly CURL_OPTIONS="-L --connect-timeout 10"
-
-# Get packages
-
-apt-get update
-apt-get install -y --no-install-recommends \
-                   autoconf \
-                   automake \
-                   build-essential \
-                   ca-certificates \
-                   cmake \
-                   extra-cmake-modules \
-                   git \
-                   libarchive-tools \
-                   libtool \
-                   nsis \
-                   pkg-config \
-                   python3-pefile \
-                   tclsh \
-                   texinfo \
-                   unzip \
-                   curl \
-                   yasm \
-                   zip
-
-if [[ "$ARCH" == "i686" ]]
-then
-  apt-get install -y --no-install-recommends \
-                    g++-mingw-w64-i686 \
-                    gcc-mingw-w64-i686
-  update-alternatives --set i686-w64-mingw32-gcc /usr/bin/i686-w64-mingw32-gcc-posix
-  update-alternatives --set i686-w64-mingw32-g++ /usr/bin/i686-w64-mingw32-g++-posix
-elif [[ "$ARCH" == "x86_64" ]]
-then
-  apt-get install -y --no-install-recommends \
-                    g++-mingw-w64-x86-64 \
-                    gcc-mingw-w64-x86-64
-  update-alternatives --set x86_64-w64-mingw32-gcc /usr/bin/x86_64-w64-mingw32-gcc-posix
-  update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
-fi
-
-
-# Install wine to run qTox tests in
-set +u
-if [ -z "$TRAVIS_CI_STAGE" ] || [[ "$TRAVIS_CI_STAGE" == "stage3" ]]
-then
-  dpkg --add-architecture i386
-  apt-get update
-  apt-get install -y wine wine32 wine64
-fi
-set -u
-
-
-# Use all cores for building
-
-MAKEFLAGS=j$(nproc)
-export MAKEFLAGS
-
-
-# Helper functions
-
-# We check sha256 of all tarballs we download
-check_sha256()
-{
-  if ! ( echo "$1  $2" | sha256sum -c --status - )
-  then
-    echo "Error: sha256 of $2 doesn't match the known one."
-    echo "Expected: $1  $2"
-    echo -n "Got: "
-    sha256sum "$2"
-    exit 1
-  else
-    echo "sha256 matches the expected one: $1"
-  fi
-}
-
-# If it's not a tarball but a git repo, let's check a hash of a file containing hashes of all files
-check_sha256_git()
-{
-  # There shoudl be .git directory
-  if [ ! -d ".git" ]
-  then
-    echo "Error: this function should be called in the root of a git repository."
-    exit 1
-  fi
-  # Create a file listing hashes of all the files except .git/*
-  find . -type f | grep -v "^./.git" | LC_COLLATE=C sort --stable --ignore-case | xargs sha256sum > "/tmp/hashes-$1.sha"
-  check_sha256 "$1" "/tmp/hashes-$1.sha"
-}
-
-# Strip binaries to reduce file size, we don't need this information anyway
-strip_all()
-{
-  set +e
-  for PREFIX_DIR in $DEP_DIR/*; do
-    strip --strip-unneeded $PREFIX_DIR/bin/*
-    $ARCH-w64-mingw32-strip --strip-unneeded $PREFIX_DIR/bin/*
-    $ARCH-w64-mingw32-strip --strip-unneeded $PREFIX_DIR/lib/*
-  done
-  set -e
-}
```

Comment:
Why do you store un-stripped binaries in the Docker image? Stripping them would reduce the size by a lot. This is what this function was doing - reducing the size of the cached dependencies, to cache them on Travis for the future builds.

Reply:
This is useful for debugging. When we have CI failures or crash dumps we can use these symbols to get more accurate backtraces in gdb. I've already used this to address a windows specific segfault in a PR. These images are <500MB, we have 10G in the github actions cache. I think this is a worthwhile tradeoff.

Reply:
I'm so confused by what you have said.

Can you give a more concrete example when this is helpful?

The symbols don't help with CI failures. CI failures are usually build failures, not runtime failures, you don't run Windows programs inside CI, aside from the qTox tests perhaps, and then you only need qTox symbols for this, which you have at the time the tests are running.

Did you perhaps mean that it's useful for the distributed debug builds, so that Windows users could provide even more detailed backtraces? If so, it could make sense to also bundle the source code of all deps for the debug distribution.